ios/KMMICTests/Sources/Specs/Presentation/Modules/SurveyQuestion/SurveyQuestionViewModelSpec.swift

```diff
@@ -35,9 +35,166 @@ final class SurveyQuestionViewModelSpec: QuickSpec {
                     title: question?.text ?? "",
                     coverImageURL: .init(string: question?.coverImageUrl ?? ""),
                     displayType: QuestionDisplayType.heart,
-                    answers: question?.answers ?? []
+                    answers: question?.answers.map {
+                        .init(answer: $0)
+                    } ?? []
                 )
             }
+
+            describe("its isLast") {
+
+                context("when it's the last question") {
+
+                    beforeEach {
+                        viewModel = SurveyQuestionViewModel(
+                            surveyDetail: .dummy,
+                            questionIndex: 0
+                        )
+                    }
+
+                    it("returns isLast correctly") {
+                        expect(viewModel.isLast) == true
+                    }
+                }
+
+                context("when it's not the last question") {
+
+                    let surveyDetail = SurveyDetail.dummy(questions: [
+                        .dummy(id: "1", displayType: .choice),
+                        .dummy(id: "2", displayType: .dropdown)
+                    ])
+
+                    beforeEach {
+                        viewModel = SurveyQuestionViewModel(
+                            surveyDetail: surveyDetail,
+                            questionIndex: 0
+                        )
+                    }
+
+                    it("returns isLast correctly") {
+                        expect(viewModel.isLast) == false
+                    }
+                }
+            }
+
+            describe("its isAllValid") {
+                let displayTypes: [QuestionDisplayType] = [.dropdown, .heart, .star, .smiley, .nps, .textarea, .choice]
+                for displayType in displayTypes {
+                    let surveyDetail = SurveyDetail.dummy(questions: [
+                        .dummy(id: "1", displayType: displayType)
+                    ])
+                    context("when displayType is \(displayType)") {
+
+                        context("when answer inputs is empty") {
+
+                            beforeEach {
+                                viewModel = SurveyQuestionViewModel(
+                                    surveyDetail: surveyDetail,
+                                    questionIndex: 0
+                                )
+                            }
+
+                            it("returns isAllValid correctly") {
+                                expect(viewModel.isAllValid) == false
+                            }
+                        }
+
+                        context("when answer inputs is not empty") {
```

Comment:
when answer inputs are not empty

Reply:
@phongvhd93 updated at [fef4e7f](https://github.com/markgravity/kmm-ic/pull/96/commits/fef4e7fbe8f7bf595a52de289ebe31252be2e656)